vitrina/structure/views.py

```diff
@@ -3424,34 +3435,212 @@ def form_valid(self, form):
+                        type=new_metadata_instance.type if new_metadata_instance.type else None,
+                        required=new_metadata_instance.required,
+                        unique=new_metadata_instance.unique,
+                        type_args=new_metadata_instance.type_args if new_metadata_instance.type_args else None,
+                        ref=new_metadata_instance.ref if new_metadata_instance.ref else None,
+                        source=new_metadata_instance.source if new_metadata_instance.source else None,
+                        prepare=new_metadata_instance.prepare if new_metadata_instance.prepare else None,
+                        level_given=new_metadata_instance.level_given,
+                        access=new_metadata_instance.access,
+                        base=new_metadata_instance.object.base
+                        if isinstance(new_metadata_instance.object, Model)
+                        else None,
+                        status=new_metadata_instance.status if new_metadata_instance.status else None,
+                    )
+
+            already_created_fields = old_to_new_metadata_object_map
+            for old_related_instance, new_related_instance in list(old_to_new_metadata_object_map.items()):
+                try:
+                    already_created_fields = self.duplicate_foreign_key_relationships(
+                        new_related_instance, already_created_fields
+                    )
+                except ValidationError as e:
+                    transaction.set_rollback(True)
+                    form.add_error(None, e.message)
+                    return self.form_invalid(form)
+
+        if self.new_version:
+            return redirect(reverse("dataset-structure", args=[self.dataset.pk, self.new_version.pk]))
+        else:
+            return redirect(reverse("dataset-structure", args=[self.dataset.pk, self.metadata_version.pk]))
+
+    def create_related_model_duplicate(self, old_metadata_instance):
+        related_model = old_metadata_instance.content_type.model_class()
+        if related_model == Dataset:
+            return None
+        old_related_instance = related_model.objects.filter(id=old_metadata_instance.object_id).first()
+        new_related_instance = deepcopy(old_related_instance)
+        new_related_instance.pk = None
+        new_related_instance.metadata_version = self.new_version
+        new_related_instance.save()
+
+        return old_related_instance, new_related_instance
+
+    def create_metadata_duplicate(self, old_metadata_instance) -> tuple:
+        new_metadata_instance = deepcopy(old_metadata_instance)
+        new_metadata_instance.pk = None
+        # TODO: column draft can be deprecated after full versioning is completed.
+        new_metadata_instance.draft = False
+        if old_metadata_instance.status == Status.objects.filter(codename=StatusCode.DEVELOP).first():
```

Comment:
I don't really like `Status.objects.filter(codename=StatusCode.DEVELOP).first()` query. It leaves a lot of assumptions:
- Why `.first()`? Can there be more than one `Status` object with `StatusCode.DEVELOP`? 
- If not - maybe this should be `Status.objects.get(codename=StatusCode.DEVELOP)` or just a value saved in `self`?
- If yes - why we only compare status to first one
- It also assumes that if `old_metadata_instance` doesn't have status and `Status` does not exist - it's also fine. Is this really the case?

Maybe it would be better to write it this way? (But I am not sure if this is correct for this use case)
```suggestion
        if old_metadata_instance.status and old_metadata_instace.status.codename == StatusCode.DEVELOP:
```

Reply:
yeah good point, the statuses are created in a migration so they should always exist.